### DISCUSSION TITLE
How to releace just the selected text

### DISCUSSION THREAD

**Discussion by shavelli** [danielsaidi/RichTextKit#205] [Q&A]
How can i replace just the selected text?

I'm just experimenting, but i'm trying to add ordered and unordered lists functionality. (I was hoping this would be included already, but I don't think it is)

Anyway i tried this but it replaces the entire text. How do i replace just the text that the user is selecting? Even better if i could keep it as NSAttributedString the whole time so i don't lose other user formatting.

Experiment Code:

```
  func toggleOrderedList(context: RichTextContext) {
        guard context.hasSelectedRange else { return }
        
        // Split the selected text into lines
        let lines = context.attributedString.string.components(separatedBy: "\n")
        
        // Add numbering to each line
        let orderedList = lines.enumerated().map { (index, line) in
            return "\(index + 1). \(line)"
        }.joined(separator: "\n")
        
        // Replace the selected text with the formatted ordered list
        context.setAttributedString(to: orderedList)

    }
```

**Comment by danielsaidi** [comment]
Hi @shavelli 

If you manage to implement lists, I'd love to see it so I can add it to the SDK.

You could try to trigger the paste action with the selected range, that should work.

Please let me know once you've tried.
